app/models/akka/QoSState.scala

```diff
@@ -0,0 +1,109 @@
+package models.akka
+
+import akka.actor.{Actor, ActorLogging, Props}
+import models.akka.Messages._
+import models.rpc.SubscriptionNotificationMessage
+
+import scala.collection.mutable
+import scala.concurrent.duration._
+
+/**
+  * actor for storing dslink - specific state as
+  * subscriptions, qos queue etc
+  *
+  * @param maxCapacity max queue size per sid
+  * @param reconnectionTime timeout for storage drop
+  */
+class QoSState(val maxCapacity: Int = 30, val reconnectionTime:Int = 30) extends Actor with ActorLogging {
+
+  var subscriptionsQueue = Map[Int, mutable.Queue[SubscriptionNotificationMessage]]()
```

Comment:
Would it be better to replace it with immutable queue and replace on changes? Seems safer.

Reply:
good point, replaced.